### PR TITLE
update eslint config and fix errors

### PR DESCRIPTION
**Pull request by karlandindrakryggen**
- Updates and cleanup eslint deps and updates lint command
- Adds react hooks to eslint config
- Fixes eslint error and warnings

### CODE REVIEW

File: public/js/app/components/Editor/EditorSection.jsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const EditorSection = ({ name, title, validationError, value, onChange }) => {
         {isOpen ? (
           <textarea id={editorId} name={name} defaultValue={value} />
         ) : (
+          // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
```
Comment (karlandindrakryggen) [inline public/js/app/components/Editor/EditorSection.jsx:32]: Would be great to have better a11y for this, but I think this is out of scope for this task right now. And the editor can also be activated with a (accessible) button, so this isn't a big a11y error.

File: public/js/app/components/FileInput.jsx [hunk 2]
```diff
@@ -32,6 +30,7 @@ const FileInput = React.forwardRef(function FileInput(props, ref) {
 
   return (
     <span className="btn-upload-file">
+      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
```
Comment (karlandindrakryggen) [inline public/js/app/components/FileInput.jsx:33]: Not sure what to do with this, disabling to rule for now.

File: server/requireRole.js [hunk 3]
```diff
@@ -83,10 +83,8 @@ module.exports.requireRole = (...roles) =>
...
-    _isAdminOfCourseSchool(courseCode, user).then(isAdminOfCourseSchool => {
-      req.session.passport.user.roles.isSchoolAdmin = isAdminOfCourseSchool
-
-      if (isAdminOfCourseSchool) return next()
-      else return next(messageHaveNotRights(lang))
-    })
+    const isAdminOfCourseSchool = await _isAdminOfCourseSchool(courseCode, user)
+    req.session.passport.user.roles.isSchoolAdmin = isAdminOfCourseSchool
+    if (isAdminOfCourseSchool) return next()
+    else return next(messageHaveNotRights(lang))
```
Comment (karlandindrakryggen) [inline server/requireRole.js:89]: Not sure here, but old impl could be buggy and this is better or at least the same?

**Review by belanglos — APPROVED** [review]
Horse job! :D 

As discussed :)

File: public/js/app/components/Editor/useToggleCKEditor.js [hunk 4]
```diff
@@ -20,16 +20,17 @@ function useToggleCKEditor(id, { onChange, onInstanceReady }) {
...
+        const editor = CKEDITOR.replace(id, CKEditorConfigs)
+        onInstanceReady && editor.on('instanceReady', ev => onInstanceReady(ev))
+        onChange && editor.on('change', ev => onChange(ev))
+      }
     }
-  }
+    // eslint rule disabled for onChange and onInstanceReady, would be great if
+    // it wasn't needed, but should hopefully not be a problem to disable it.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [id, isOpen])
```
[the remark this answers is not in the document]
Comment (belanglos) [reply public/js/app/components/Editor/useToggleCKEditor.js:33]: Thanks for the discussion :)

File: public/js/app/client-context/addClientFunctionsToWebContext.jsx [hunk 5]
```diff
@@ -3,7 +3,7 @@
 import axios from 'axios'
 import { safeGet } from 'safe-utils'
 
-const paramRegex = /\/(:[^\/\s]*)/g
```
[the remark this answers is not in the document]
Comment (belanglos) [reply public/js/app/client-context/addClientFunctionsToWebContext.jsx:6]: Eslint is smart...

File: public/js/app/components/FileInput.jsx (hunk 2 above)
Comment (karlandindrakryggen) [reply public/js/app/components/FileInput.jsx:33]: I'll have a closer look at this when I updating style of the button to kth-style-10
